src/conversions/cart.cpp

```diff
@@ -112,7 +112,7 @@ static double normal_radius_of_curvature(double a, double es, double sinphi) {
 }
 
 /*********************************************************************/
-static double geocentric_radius(double a, double b, double cosphi,
+static double geocentric_radius(double a, double b_div_a, double cosphi,
```

Comment:
Nice trick using `b/a`, that is always `<1`, to avoid overflows and improve performance.

Reply:
Note that `b/a` is also called the *aspect ratio* of the ellipsoid, which may be more descriptive than `b_div_a`. I have noted another pair of identities [here](https://docs.rs/geodesy/latest/geodesy/struct.Ellipsoid.html#method.aspect_ratio). There's no reference to literature, but I remember adding the name and the referred ellipsoid method right after having seen it mentioned in an absolutely reliable source (perhaps Rapp, OSU)

Reply:
Thanks. Do you know if there is a conventional short name/letter for the aspect ratio?

Reply:
> Thanks. Do you know if there is a conventional short name/letter for the aspect ratio?

I do not remember one, and as I was too stupid to write down the reference in the [Rust Geodesy Bibliography](https://docs.rs/geodesy/latest/geodesy/enum.Bibliography.html) it will be hard to find again - but [this](https://www.researchgate.net/publication/32180639_Effective_stress_and_pore_pressure_in_the_Nankai_accretionary_prism_off_the_Muroto_Peninsula_Southwestern_Japan/figures?lo=1) paper uses alpha (and swaps the meaning of a and b!), while [Rapp](https://kb.osu.edu/bitstreams/ee6892dd-cfb7-5142-978a-a978ae792980/download) uses alpha for the *angular eccentricity*, and hence `b/a = arccos(alpha)`. So it does not look like there is any firmly established convention.

Also, I see from [this](https://en.wikipedia.org/wiki/Aspect_ratio) Wikipedia page, that it is also common to define the aspect ratio as the larger parameter divided by the smaller, so even in that aspect (!) there does not seem to be any clear convention.

Reply:
after all, isn't b_div_a quite explicit ;-) ?

Reply:
> after all, isn't b_div_a quite explicit ;-) ?

It can probably harder get any more explicit :-) so given the lack of common nomenclature, it is obviously the better choice!

Reply:
> Nice trick using `b/a`, that is always `<1`, to avoid overflows and improve performance.

... although I think a few of Jupiter's moons are actually prolate, so this will not hold if we at some time accidentally blindly import IAU ellipsoids

Reply:
The algorithm is still stable. If it were thaaat eccentric, then using an ellipsoid model wouldn't be a good idea ;)
I'm curious what is the physics behind a spinning prolate moon.

Reply:
> The algorithm is still stable.

Ah, yes, I now see that it is, since we may safely assume that b never equals the imaginary unit times a :-)

> I'm curious what is the physics behind a spinning prolate moon.

Physically, I believe it's as stable as an oblate, as long as the spin axis and the principal axes inertia are reasonably aligned. Apparently it just happens to spin around the longest axis, for whichever reason `$DEITY` came up with during the creation of the solar system (and, in the case of Saturn, whichever perturbations the poor moon suffers from the debris of the ring it shepherds)